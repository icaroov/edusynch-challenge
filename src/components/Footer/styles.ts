import styled, { css } from 'styled-components'

export const Container = styled.footer`
  display: flex;
  justify-content: center;
  align-items: center;
`

export const Text = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.black};
    font-size: ${theme.font.sizes.small};

    text-align: center;
  `}
`
