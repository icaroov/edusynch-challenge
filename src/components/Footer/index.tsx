import * as Styled from './styles'

const Footer = () => (
  <Styled.Container>
    <Styled.Text>
      Copyright &copy; 2021 <strong>Eduick</strong>. Todos os direitos
      reservados.
    </Styled.Text>
  </Styled.Container>
)

export default Footer
