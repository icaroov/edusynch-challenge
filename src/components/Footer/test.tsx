import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Footer from '.'

describe('<Footer />', () => {
  it('should render the footer correctly', () => {
    const { container } = renderWithTheme(<Footer />)

    expect(screen.getByText(/Eduick/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
