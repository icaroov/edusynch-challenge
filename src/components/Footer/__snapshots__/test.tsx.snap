// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render the footer correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  color: #030517;
  font-size: 1.4rem;
  text-align: center;
}

@media screen and (max-width:768px) {
  .c1 {
    font-size: 1rem;
  }
}

<footer
  class="c0"
>
  <p
    class="c1"
  >
    Copyright Â© 2021 
    <strong>
      Eduick
    </strong>
    . Todos os direitos reservados.
  </p>
</footer>
`;
